client/my-sites/sharing/connections/service.jsx

```diff
@@ -1,108 +1,184 @@
-			isRefreshing: false     // A pending refresh is awaiting completion
+			isRefreshing: false,    // A pending refresh is awaiting completion
+			isSelectingAccount: false,
 		};
 	},
 
-	getDefaultProps: function() {
-		return {
-			onAddConnection: function() {},
-			onRemoveConnection: function() {},
-			onRefreshConnection: function() {},
-			onToggleSitewideConnection: function() {}
-		};
+	componentWillReceiveProps: function( nextProps ) {
+		if ( this.getConnections().length !== nextProps.siteUserConnections.length ) {
+			this.setState( {
+				isConnecting: false,
+				isDisconnecting: false,
+				isSelectingAccount: false,
+			} );
+		}
 	},
 
 	componentWillUnmount: function() {
-		this.props.connections.off( 'create:success', this.onConnectionSuccess );
-		this.props.connections.off( 'create:error', this.onConnectionError );
 		this.props.connections.off( 'destroy:success', this.onDisconnectionSuccess );
 		this.props.connections.off( 'destroy:error', this.onDisconnectionError );
 		this.props.connections.off( 'refresh:success', this.onRefreshSuccess );
 		this.props.connections.off( 'refresh:error', this.onRefreshError );
 	},
 
-	onConnectionSuccess: function() {
-		this.setState( { isConnecting: false } );
-		this.props.connections.off( 'create:error', this.onConnectionError );
-
-		notices.success( this.translate( 'The %(service)s account was successfully connected.', {
-			args: { service: this.props.service.label },
-			context: 'Sharing: Publicize connection confirmation'
-		} ) );
-
-		if ( ! this.state.isOpen ) {
-			this.setState( { isOpen: true } );
-		}
-	},
-
-	onConnectionError: function( reason ) {
-		this.setState( { isConnecting: false } );
-		this.props.connections.off( 'create:success', this.onConnectionSuccess );
-
-		if ( reason && reason.cancel ) {
-			notices.warning( this.translate( 'The %(service)s connection could not be made because no account was selected.', {
-				args: { service: this.props.service.label },
-				context: 'Sharing: Publicize connection confirmation'
-			} ) );
-		} else if ( reason && reason.connected ) {
-			notices.warning( this.translate( 'The %(service)s connection could not be made because all available accounts are already connected.', {
-				args: { service: this.props.service.label },
-				context: 'Sharing: Publicize connection confirmation'
-			} ) );
-		} else {
-			notices.error( this.translate( 'The %(service)s connection could not be made.', {
-				args: { service: this.props.service.label },
-				context: 'Sharing: Publicize connection confirmation'
-			} ) );
-		}
-	},
-
 	onDisconnectionSuccess: function() {
 		this.setState( { isDisconnecting: false } );
 		this.props.connections.off( 'destroy:error', this.onDisconnectionError );
 
-		notices.success( this.translate( 'The %(service)s account was successfully disconnected.', {
+		this.props.successNotice( this.props.translate( 'The %(service)s account was successfully disconnected.', {
```

Comment:
If there's are appropriate redux action types to associate this with, adding these to notice middleware may make sense: https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/middleware.js